### DISCUSSION TITLE
✅ TOC widget: ability to customize "No headings were found on this page." fallback message

### DISCUSSION THREAD

**Discussion by joehdevwp** [elementor/elementor#21904] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

In cases where the TOC widget does not find any headings, it displays the phrase "No headings were found on this page."  This fallback phrase is not editable/customizeable. 

### Describe the solution you'd like

The TOC widget would be even more useful and more versatile if we could change this fallback phrase.  For example, "there are no events at this time, please check back later". 

Because the default phrase is not always appropriate, we have to find ways to hide the TOC widget when it has no headings due to content on the page being loaded conditionally. 

### Describe alternatives you've considered

Using javascript to search the DOM for specific headings and if none are found to hide the TOC widget entirely.

**Comment by toineenzo** [comment]
I second this. It should also be possible to hide it. It's pretty useless to show the TOC when the post content doesnt contain any headings

**Comment by nicholaszein** [comment]
Hello there @joehdevwp!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor Pro v3.25.0! 🥳 

You can now write a customized message when no headings are found in the Table of Contents widget.

![image](https://github.com/user-attachments/assets/2af22ec7-3cf3-42f1-a750-809f6a6e5042)


✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
